### PR TITLE
[OpenMP][Fix] libomptarget Fortran tests

### PR DESCRIPTION
**Pull request by fabianmcg**
This patch fixes the erroneous multiple-target requirement in Fortran offloading tests. Additionally, it adds two new variables (`test_flags_clang`, `test_flags_flang`) to `lit.cfg` so that compiler-specific flags for Clang and Flang can be specified.

### CODE REVIEW

File: openmp/libomptarget/test/offloading/fortran/constant-arr-index.f90 [hunk 1]
```diff
@@ -2,8 +2,7 @@
 ! that checks constant indexing on device
 ! correctly works (regression test for prior
 ! bug).
-! REQUIRES: flang, amdgcn-amd-amdhsa
-! UNSUPPORTED: nvptx64-nvidia-cuda
+! REQUIRES: flang
 ! UNSUPPORTED: nvptx64-nvidia-cuda-LTO
```
Comment (jdoerfert) [inline openmp/libomptarget/test/offloading/fortran/constant-arr-index.f90:6]: why is nvptx still unsupported?

File: openmp/libomptarget/test/lit.cfg [hunk 2]
```diff
@@ -136,7 +140,7 @@ else: # Unices
     if config.cuda_libdir:
         config.test_flags += " -Wl,-rpath," + config.cuda_libdir
     if config.libomptarget_current_target.startswith('nvptx'):
-        config.test_flags += " --libomptarget-nvptx-bc-path=" + config.library_dir + '/DeviceRTL'
```
Comment (shiltian) [inline openmp/libomptarget/test/lit.cfg:139]: I think the leading whitespace needs to stay in case `config.test_flags_clang` is not empty.

**Review by shiltian — APPROVED** [review]
LGTM

**Review by jdoerfert — APPROVED** [review]
LG, the cuda test support can be a follow up.

**Comment by jplehr** [comment]
This patch seems to have made the AMDGPU OpenMP Offload buildbot unhappy (https://lab.llvm.org/buildbot/#/builders/193/builds/44036).

If you need help in recreating the issue, let me know.

**Comment by fabianmcg** [comment]
I've just reverted, I'll investigate the issue offline and try to replicate.

**Pull request merged**